arch/risc-v/src/mpfs/mpfs_i2c.c

```diff
@@ -692,6 +699,41 @@ static int mpfs_i2c_transfer(struct i2c_master_s *dev,
   return ret;
 }
 
+/****************************************************************************
+ * Name: mpfs_i2c_reset
+ *
+ * Description:
+ *   Performs an I2C bus reset. This may be used to recover from a buggy
+ *   situation.
+ *
+ * Input Parameters:
+ *   dev   - Device-specific state data
+ *
+ * Returned Value:
+ *   Zero (OK) on success; this should not fail.
+ *
+ ****************************************************************************/
+#ifdef CONFIG_I2C_RESET
+static int mpfs_i2c_reset(struct i2c_master_s *dev)
```

Comment:
@jrosberg - I was expecting some [IO manipulation](https://github.com/apache/incubator-nuttx/blob/master/arch/arm/src/stm32/stm32f40xxx_i2c.c#L2514-L2634) does this HW do a IO line reset automatically?

Reply:
@davids5 some MCU has support to it in hardware, ESP32-C3 is another example of it.

Reply:
@acassis that is cool! I see @saramonteiro just changed it over. Nice.